docs/design/proposals/rbd-mirror.md

```diff
@@ -0,0 +1,96 @@
+# RBD MIRRORING
+
+RBD mirroring is a process of replication of RBD images between two or more
+Ceph clusters. Mirroring ensures point-in-time, crash-consistent RBD images
+between clusters, RBD mirroring is mainly used for disaster recovery (i.e.
+having a secondary site as a failover). See [Ceph
+documentation](https://docs.ceph.com/en/latest/rbd/rbd-mirroring) on RBD
+mirroring for complete information.
+
+## Architecture
+
+![mirror](rbd-mirror.png)
+
+## Design
+
+Currently, CEPH-CSI generates its unique ID for each RBD image and stores the
+corresponding PersistentVolume (PV) name and the unique ID mapping, It creates
+the RBD image with the unique ID and returns the encoded value which contains
+all the required information for other operations. For mirroring, the same RBD
+image will be mirrored to the secondary cluster. As the journal(OMAP data) is
+not mirrored to the secondary cluster, The RBD images corresponding to the PV
+can not be identified without OMAP data.
+
+**Pre-req** It's expected that the Kubernetes Admin/User will create the static
+PersistentVolumeClaim (PVC) on the secondary site during the failover.
+
+**Note:** when the static PVC created on the secondary site we cannot use the
+VolumeHandle to identify the OMAP data nor the image anymore because as we have
+only PoolID and ClusterID in the VolumeHandle. we cannot identify the correct
+pool name from the PoolID because pool name will remain the same on both
+clusters but not the PoolID even the ClusterID can be different on the
+secondary cluster.
+
+> Sample PV spec which will be used by controller to regenerate OMAP data
```

Comment:
*.. by controller * - which controller? If it is the **Replication Controller** in the diagram, reference to it by full name (everywhere).

If one of the components that will be developed for this design is this controller, it can be explained a little more as one of the deliverable components. Maybe it is good to list which components need to get developed from scratch, and which ones need some modifications/extensions.

Reply:
done added more details, we dont be designing the Replication Controller as of now, this will be a new controller which will run as a sidecar in the provisioner pod